File: src/models/case.model.js
const { Schema, model } = require('mongoose');

const caseSchema = new Schema({
    name: { type: String, required: true },
    brand: { type: String, required: true },
    form_factor: { type: String, enum: ['ATX', 'Micro-ATX', 'Mini-ITX'], required: true },
    max_gpu_length_mm: { type: Number, required: true },
    max_cooler_height_mm: { type: Number, required: true },
    url: { type: String, require: true },
    price: { type: Number, required: true}
});

module.exports = model('case', caseSchema);

File: lib/models/case.model.dart
class Case {
  final String urlImage;
  final String name;
  final String brand;
  final String formFactor;
  final int maxGpuLengthMm;
  final int maxCoolerHeightMm;
  final String url;
  final int price;

  Case({
    required this.urlImage,
    required this.name,
    required this.brand,
    required this.formFactor,
    required this.maxGpuLengthMm,
    required this.maxCoolerHeightMm,
    required this.url,
    required this.price
  });

  factory Case.fromJson(Map<String, dynamic> json) {
    return Case(
      urlImage: json['url_image'],
      name: json['name'],
      brand: json['brand'],
      formFactor: json['formFactor'],
      maxGpuLengthMm: json['maxGpuLengthMm'],
      maxCoolerHeightMm: json['maxCoolerHeightMm'],
      url: json['url'],
      price: json['price'],
    );
  }
}
